### DISCUSSION TITLE
Symfony v6 support

### DISCUSSION THREAD

**Discussion by driesvints** [php-webdriver/php-webdriver#924] [Ideas]
I know it's still early but is there a chance we can start discussing how Symfony v6 will be tackled in this repo? I unfortunately don't know much of the internals to know all the implications. We might be able to support it in the current stable branch but if Symfony's new type adaptations is blocking that then a new major version will need to be released.

The reason why I'm asking this is because we're currently trying to get a develop branch for Dusk to build with Symfony v6 support (required in the upcoming Laravel v9 release) and php-webdriver is a blocker for us atm: https://github.com/laravel/dusk/pull/906

Like I said: nothing is urgent here yet. Just wanted to kickstart the discussion 🙂

**Comment by driesvints** [comment]
@OndraM @williamdes I'm sorry to ping you but is this the correct place to ask this? Do you want me to create an issue instead? Let me know if I can help out here!

**Reply by williamdes** [reply]
An issue is probably a better idea, that said there is a button to "Create issue from discussion" so I will let someone else try it out if it makes sense ^^

About the support I see no reason to not support v6
https://github.com/php-webdriver/php-webdriver/blob/da16e39968f8dd5cfb7d07eef91dc2b731c69880/composer.json#L20

**Reply by driesvints** [reply]
Okay thanks @williamdes, I'll let @OndraM convert it to an issue if he wants to. The reason I choose this place was because https://github.com/php-webdriver/php-webdriver/issues/new/choose indicates new ideas and feature requests should be posted here and I didn't really consider this to be a bug. I'll wait until @OndraM has had time to reply! Nothing super urgent here :)

**Comment by OndraM** [comment]
Hi,
sure it is - I'm currently out of office until next week, so I will have a
look then.


Have a great day!
OM



> @OndraM <https://github.com/OndraM> @williamdes
> <https://github.com/williamdes> I'm sorry to ping you but is this the
> correct place to ask this? Do you want me to create an issue instead? Let
> me know if I can help out here!

**Comment by OndraM** [comment]
Well, from what I've found, there don't seem to be BC breaks in Symfony/Process component.

However there is dev-dependency on php-cs-fixer, which requires Symfony 5.0 (even with version 3.0, which we don't use...). I did a quick test with removed php-cs-fixer, and it seems to be working with tests passing.

So - the question is what to do next:
- First, we should upgrade php-cs-fixer to 3.0, and remove it from the main composer.json to `tools` directory, so that its dependencies won't conflict future updates. I will prepare issue for this and will be happy for anyone volunteering on resolving this.
- Then I think it would be safe to allow 6.0 process and var-dumper, and hopefully include this in some minor php-webdriver release.

Any other thoughts or ideas on this?

**Reply by williamdes** [reply]
It looks like a good plan to me

**Reply by driesvints** [reply]
Great! :)

**Reply by OndraM** [reply]
PR with update to php-cs-fixer 3 is here: https://github.com/php-webdriver/php-webdriver/pull/930

**Comment by OndraM** [comment]
I created a branch `feature/symfony-6` with Symfony 6 support, and the builds are all passing. :ok_hand: You can give it a try @driesvints :rocket: . And if everything will be ok, we can marge it to the main branch.

https://packagist.org/packages/php-webdriver/webdriver#dev-feature/symfony-6

**Reply by driesvints** [reply]
Seems to work great! 🎉 

https://github.com/laravel/dusk/commit/bcdd399290ddaaed1963f6498ce91940802911d1

**Reply by OndraM** [reply]
Ok, thanks for verifying. I made a proper PR here #931 to get in in `main` branch soon.

**Reply by driesvints** [reply]
Thanks for that! 🙂

**Reply by OndraM** [reply]
Merged into the `main` branch :rocket:
